File: Team5/php/rasa.php
<?php
$totalLines = 0;
$totalRasa = 0;
for($i = 1; isset($argv[$i]); $i++) {
    $file = $argv[$i];
    $lines =  count(file($file));
    $rasa = rasa(file_get_contents($file), 1);
    $totalLines += $lines;
    $totalRasa += $rasa;
    echo "$file: lines " . $lines . ", RaSa " . $rasa ."\n";

}
echo "total: lines $totalLines, RaSa $totalRasa \n";

function rasa($code, $depth)
{
    $stms = 0;
    $rasa = 0;

    $chars = str_split($code);
    for($i = 0; $i < strlen($code); $i++) {
        $c = $chars[$i];
        if($c === ";") {
            $stms++;
        } elseif($c === "{") {

            $block = readBlock(substr($code, $i));
            $i += strlen($block);
            $rasa += rasa($block, $depth + 1);

        } 
    }
    return $rasa + $stms * $depth;
}

function readBlock($input)
{
    $stack = [];
    $chars = str_split($input);
    $char = current($chars);
    $output = "";
    do {
        if($char === '{') {
            array_push($stack, $char);
        } elseif($char === '}') {
            array_pop($stack);
        }
        $output .= $char;
        $char = next($chars);
    } while(!empty($stack) && $char !== false);
    return substr($output, 1, strlen($output) - 2);

}

File: Team5/cpp/rasa.cpp
#include <stdio.h>
#include <iostream>
#include "fstream"
#include "string"
using namespace std;
int main (int argc, char *argv[])
{
 int depth = 1;
 int summaryRasa = 0;
 int totalLines = 0;

 for (i = 1; i < argc; i++) {
 	ifstream fin(argv[i], std::ios::in);
 	int rasa = 0;
 	int lines = 0;
	 while (1) {
	 	char ch = fin.get();
	 	if (fin.eof())
	 		break;
	 	switch (ch) {
	 		case ';': rasa += depth; break;
	 		case '{': depth ++; break;
	 		case '}': depth --; break;
	 		case '\n': lines ++; break;
	 	}
	 }
	 std::cout << argv[i] << ": lines "<<lines<<", RaSa "<<rasa<<"\n";
	 summaryRasa += rasa;
	 totalLines += lines;
 }
 
 std::cout << "total: lines "<<totalLines<<", RaSa "<<summaryRasa<<"\n";
}